include/oneapi/dpl/pstl/hetero/numeric_ranges_impl_hetero.h

```diff
@@ -93,6 +93,7 @@ __pattern_transform_scan_base(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Rang
 {
     if (__rng1.empty())
         return 0;
+    oneapi::dpl::__internal::__difference_t<_Range2> __rng1_size = __rng1.size();
```

Comment:
Should this type be `oneapi::dpl::__internal::__difference_t<_Range1>`?

Reply:
I'm fine with it as is or with `auto`, which would convert during the return (but would make the code less verbose).